### ISSUE TITLE
Encryption

### ISSUE THREAD

**Issue by dfahlander** [dexie/Dexie.js#1861]
Feature Request

It is already possible to do encryption with dexie using [dexie-encrypted](https://github.com/mark43/dexie-encrypted) and [dexie-easy-encrypt](https://github.com/jaetask/dexie-easy-encrypt). However, these library are not actively maintained.

This feature request is about adding an encryption addon maintained within this repo.

**Comment by nlaurie** [comment]
I'm in the process of rewriting dexie-encrypted ,  is it is way out of date.  I have since upgraded it to ESM and fixed all the indexing issues.  Currently working through one more issue with the distinct().  Happy to bring it into the core.

**Comment by snyappy** [comment]
@nlaurie Hi, the dexie-encrypted github page was removed. So you're currently working on integrating encryption to dexie itself?

**Comment by nlaurie** [comment]
I have converted the project to ESM and fixed some of the issues with it working with the latest dexie.  Since our product depends on solid encryption we will be moving this project forward, either on a branch or in the core , if David wants in included.

**Comment by dfahlander** [comment]
Thanks @nlaurie ! Do you have a link to a repository where we will find it?

**Comment by nlaurie** [comment]
We were working with the assumption it was an orphaned project ,  so I
cloned it, created a branch and started refactoring the project to work
with the latest dexie.
https://github.com/metafig/dexie-encrypted/tree/ESM

I would be happy to manage the project on our branch or yours.   Since the
base branch change might need to be hand merged ? to get it all back in sync

The plugin initialization is a work in progress due to typing issues.

Nick

**Comment by dfahlander** [comment]
That's true. The original repo seem to be deleted so my fork might be the only track we have of it. I don't see any reason to keep it in my repo though since I haven't done much on it else than one commit in a branch that I could never compile due to missing preconditions or something. And I don't have access to publish the npm package. Pinging @stutrek - would it be doable to give @nlaurie or me access to publish new versions of the package?

**Comment by stutrek** [comment]
I just invited you to be a maintainer, @dfahlander

**Comment by dfahlander** [comment]
Thanks @stutrek! I've accepted the invitation. Thanks also for all your work with the repo <3. Are you be with us taking over it? @nlaurie Let's keep in touch over how to handle it going forward.

**Comment by stutrek** [comment]
It’s all yours! I trust you’ll both do good things.

**Comment by nlaurie** [comment]
Happy too, we have a lot riding on encryption.  

Nick
